ext/lib/ipc/open-amp/CMakeLists.txt

```diff
@@ -6,6 +6,16 @@
 
 set(WITH_ZEPHYR 1)
 set(WITH_ZEPHYR_LIB 1)
+if (CONFIG_OPENAMP_MASTER)
+  set(WITH_VIRTIO_MASTER 1)
+else()
+  set(WITH_VIRTIO_MASTER OFF CACHE BOOL "" FORCE)
+endif()
+if (CONFIG_OPENAMP_SLAVE)
+  set(WITH_VIRTIO_SLAVE 1)
+else()
+  set(WITH_VIRTIO_SLAVE OFF CACHE BOOL "" FORCE)
+endif()
```

Comment:
Could you simplify by suppressing intermediate definitions (WITH_VIRTIO_MASTER , WITH_VIRTIO_SLAVE)?
Just having something like this:
if (NOT CONFIG_OPENAMP_MASTER)
  zephyr_compile_definitions(-DVIRTIO_SLAVE_ONLY)
endif()
if (NOT CONFIG_OPENAMP_SLAVE)
  zephyr_compile_definitions(-DVIRTIO_MASTER_ONLY)
endif()
if (NOT CONFIG_OPENAMP_MASTER
    AND NOT CONFIG_OPENAMP_MASTER
message(FATAL_ERROR "at least  OPENAMP_MASTER or OPENAMP_MASTER configs should be defined)
endif()

Reply:
This is moved to here: https://github.com/zephyrproject-rtos/open-amp/pull/5